{"id":"core\/lock\/locktrait","type":"","title":"Google\\Cloud\\Core\\Lock\\LockTrait","name":"LockTrait","description":"<p>Utility trait for locks.<\/p>","examples":[],"resources":[],"methods":[{"id":"acquire","type":"instance","name":"acquire","source":"src\/Core\/Lock\/LockTrait.php#L31","description":"<p>Acquires a lock that will block until released.<\/p>","examples":[],"resources":[],"params":[],"exceptions":[{"type":"\\RuntimeException","description":""}],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.boolean.php\" target=\"_blank\">bool<\/a>"],"description":""}]},{"id":"release","type":"instance","name":"release","source":"src\/Core\/Lock\/LockTrait.php#L38","description":"<p>Releases the lock.<\/p>","examples":[],"resources":[],"params":[],"exceptions":[{"type":"\\RuntimeException","description":""}],"returns":[]},{"id":"synchronize","type":"instance","name":"synchronize","source":"src\/Core\/Lock\/LockTrait.php#L48","description":"<p>Execute a callable within a lock. If an exception is caught during\nexecution of the callable the lock will first be released before throwing\nit.<\/p>","examples":[],"resources":[],"params":[{"name":"func","description":"<p>The callable to execute.<\/p>","types":["<a href=\"http:\/\/php.net\/manual\/en\/language.types.callable.php\" target=\"_blank\">callable<\/a>"],"optional":false,"nullable":null}],"exceptions":[],"returns":[{"types":["<a href=\"http:\/\/php.net\/manual\/en\/language.pseudo-types.php#language.types.mixed\" target=\"_blank\">mixed<\/a>"],"description":""}]}]}